### ISSUE TITLE
Getting TLE Fibonacci Sequence (Harder)

### ISSUE THREAD

**Issue by mahbd** [Maillew/DMOJ---Questions#1]
I copied and submitted your code Fibonacci Sequence (Harder) directly and getting TLE. Can you please fix this?

**Comment by Maillew** [comment]
Subject: Addressing Time Limit Exceeded Issue with Fibonacci Sequence Algorithm

Dear mahbd,

Thank you for reaching out and bringing this issue to our attention regarding the Fibonacci Sequence (Harder) code. I appreciate your patience and willingness to collaborate on resolving this matter.

Upon reviewing your concern, it's evident that the code is encountering a Time Limit Exceeded (TLE) error. This issue typically arises when the algorithm's efficiency is compromised, leading to excessive computation time, especially for larger input values of n. In the case of the Fibonacci sequence, employing a naive recursive approach, as demonstrated in the original code, can indeed result in inefficient execution due to redundant calculations.

To address this challenge effectively, it's imperative to delve deeper into the intricacies of the Fibonacci sequence computation and explore optimization techniques that enhance the algorithm's efficiency. By optimizing the code, we aim to mitigate the TLE error and ensure smoother execution across various input scenarios.

One of the fundamental optimizations for the Fibonacci sequence algorithm involves the integration of memoization, a technique that stores previously computed results to avoid redundant calculations. By leveraging memoization, we can significantly reduce the computational overhead associated with repetitive Fibonacci number calculations. In essence, this approach optimizes the algorithm's time complexity, transforming it from exponential to linear, thereby improving overall performance.

Moreover, while memoization serves as a valuable optimization strategy, it's essential to consider alternative approaches, particularly for scenarios involving exceptionally large input values of n. In such cases, an iterative solution often proves to be more efficient than the recursive counterpart. Iterative algorithms eliminate the overhead associated with function calls and recursion, resulting in faster execution and better scalability, especially when dealing with substantial input sizes.

By incorporating these optimization techniques, we can enhance the efficiency and performance of the Fibonacci sequence algorithm, thereby addressing the TLE error effectively. However, it's crucial to strike a balance between computational efficiency and code simplicity, ensuring that the solution remains maintainable and comprehensible for future iterations and collaborations.

Moving forward, I propose exploring these optimization strategies and implementing the necessary changes to the codebase. By iteratively refining the algorithm and conducting rigorous testing, we can achieve robustness and reliability, thereby delivering a solution that meets the required performance standards.

Once again, I sincerely appreciate your engagement and collaboration in resolving this issue. Your feedback and contributions are invaluable in our pursuit of excellence. Should you have any further questions or concerns, please don't hesitate to reach out. Together, we can overcome this challenge and elevate the quality of our codebase.

Looking forward to your insights and collaboration.

Warm regards,

Maillew

**Comment by sjay05** [comment]
Yeah BigInt won't work here. You need to model the linear recurrence with matrices @Maillew.

**Comment by Maillew** [comment]
Looking forward to your response. I hope we can collaborate some time @mahbd 🤝🤝🤝

**Comment by kevinyvv** [comment]
> @mahbd

Hello, I also ran into the same issue. But now @Maillew has fixed it, and we can rejoice.
